### PR TITLE
Implement script to measure network performance

### PR DESCRIPTION
**Pull request by mzampetakis**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,78 @@
-# netper
\ No newline at end of file
+# Network Performance
+
+This repo hosts a shell script that will run various network performance measurements using various configuration
```
Comment (sonaht) [inline README.md:3]: This repo hosts a shell script that will run various network performance measurements using variable configuration

File: README.md [hunk 2]
```diff
@@ -1 +1,78 @@
...
+| `PING_PACKET_SIZE`    | Packet size of packets send by ping tests. (`-i` Arg) (Multiple args delimited by `;`)   | "56"           | "32;56;128;1024;4096"               |
+| `PING_LATENCY`        | Add custom latency to network.                                                           | "0ms"          | "5ms;50ms;100ms"                    |
+
+> EnvVars marked with a `*` are mandatory to run the tests.
+
+The `.env.sample` file contains the suggested variation values. Renaming it to `.env` will cause it to be loaded and 
+used by the script when it runs. Feel free to ignore it, if you are using another way to set EnvVars. For cases that no 
+configuration is provided the script will run the iperf and ping measurements using the default values.
+
+For the Name variable it is recommended to add a json formatted string to contain all the details of the hardware setup
```
Comment (sonaht) [inline README.md:41]: I would suggest an example to clarify the Name variable. Maybe also indicate the issue of what happens when no Name is provided or Name is not altered on consecutive variation runs

File: README.md (hunk 1 above)
Comment (sonaht) [inline README.md]: I suggest to write it this way "This repo hosts a shell script that will run various network performance measurements using variable configuration"

File: README.md [hunk 3]
```diff
@@ -1 +1,78 @@
...
+The `.env.sample` file contains the suggested variation values. Renaming it to `.env` will cause it to be loaded and 
+used by the script when it runs. Feel free to ignore it, if you are using another way to set EnvVars. For cases that no 
+configuration is provided the script will run the iperf and ping measurements using the default values.
+
+For the Name variable it is recommended to add a json formatted string to contain all the details of the hardware setup 
+in order to allow later to identify the setup.
+
+## Running the script
+
+Before proceeding with executing the script, ensure that the iperf3 server is running on the other host using:
```
Comment (sonaht) [inline README.md]: Choose to say host 1 and host 2, or server and client. Avoid "other"

File: README.md [hunk 4]
```diff
@@ -1 +1,78 @@
...
+```bash
+iperf3 -s -p ${IPERF_PORT} -J
+```
+
+In order to run the script we must set first any configuration through EnvVars and then execute it by running:
+```bash
+sudo ./netperf.sh
```
Comment (sonaht) [inline README.md:57]: Clearly indicate that this is run on the client !!

File: README.md [hunk 5]
```diff
@@ -1 +1,78 @@
...
+```
+
+> sudo is required as the script will try to install all the required packages for the required tools.
+
+We can also set the EnvVars and run the script at the same time using
+```bash
+sudo ENVKEY1=ENVVAR1 ENVKEY2=ENVVAR2 ENVKEY3=ENVVAR3 ./netperf.sh
```
Comment (sonaht) [inline README.md]: I like this, but i would prefer an actual example with real values

File: README.md [hunk 6]
```diff
@@ -1 +1,78 @@
...
+sudo ./netperf.sh 2>&1 | tee netperf.out
+```
+
+### Results
+
+All results are located under the give `${OUTPUT}` directory (of `netperf` if none provided) and then a directory with 
+the start date and time that contains all the results. All iperf results are prefixed with `iperf` and all ping test 
+results are prefixed with `ping`. The name of each file contains all the arguments given to the command. A file with 
+the same name but with a `.cmd` extension will be also stored in the same directory and will contain the actual 
+command(s) that were executed.
```
Comment (sonaht) [inline README.md:78]: Please provide an example with actual directory listings of a run

**Review by sonaht — COMMENTED** [review]
simple things

File: README.md (hunk 2 above)
Comment (mzampetakis) [reply README.md:41]: Added a sensible example.

**Pull request merged**